packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -605,21 +606,27 @@ export class TreeCheckout implements ITreeCheckoutFork {
 					revertible.dispose();
 				}
 			},
-			clone: (forkedBranch: TreeBranch) => {
-				if (forkedBranch === undefined) {
-					return this.createRevertible(revision, kind, checkout, onRevertibleDisposed);
-				}
-
+			clone: (targetBranch: TreeBranch) => {
 				// TODO:#23442: When a revertible is cloned for a forked branch, optimize to create a fork of a revertible branch once per revision NOT once per revision per checkout.
-				const forkedCheckout = getCheckout(forkedBranch);
+				const targetCheckout = getCheckout(targetBranch);
+
 				const revertibleBranch = this.revertibleCommitBranches.get(revision);
-				assert(
-					revertibleBranch !== undefined,
-					0xa82 /* change to revert does not exist on the given forked branch */,
-				);
-				forkedCheckout.revertibleCommitBranches.set(revision, revertibleBranch.fork());
+				if (revertibleBranch === undefined) {
+					throw new UsageError("Change to revert does not exist on the branch.");
+				}
+
+				const commitToRevert = revertibleBranch.getHead();
+				const activeBranch = targetCheckout.#transaction.activeBranch;
+
+				if (findCommitInActiveBranch(commitToRevert, activeBranch) === false) {
+					throw new UsageError(
+						"Cannot clone revertible: branch A and branch B must be related",
```

Comment:
This is not the error message I suggested in my comment.
```suggestion
						"Cannot clone revertible for a commit that is not present on the given branch.",
```